src/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -221,9 +236,22 @@ protected function filterProperty(string $property, $value, QueryBuilder $queryB
         $associationResourceClass = $metadata->getAssociationTargetClass($field);
         $associationMetadata = $this->getClassMetadata($associationResourceClass);
         $associationFieldIdentifier = $associationMetadata->getIdentifierFieldNames()[0];
+        $associationResourceApiIdField = $associationFieldIdentifier;
+
+        // dig into the subResource metadata to find out which field we're looking at
+        if ($this->isResourceClass($associationResourceClass) && $this->resourceMetadataFactory) {
+            $associationApiMetadata = $this->resourceMetadataFactory->create($associationResourceClass);
+            $variables = $associationApiMetadata->getIterator()->current()->getUriVariables();
+            if (1 === \count($variables)) { // otherwise let's just give up at this point, too complicated
+                $varName = array_key_first($variables);
+                $associationResourceApiIdField = $variables[$varName]->getIdentifiers()[0]; // this will be needed to get the correct value
```

Comment:
agreed this is too complicated and deserves its own search filter...

Reply:
Yes, with multiple identifiers this becomes even clumsier to handle, and i'm not even sure what the request should look like anyway.

Reply:
Best would be to make some mix with https://github.com/api-platform/core/pull/5732 to handle complicated stuff not sure how it'd apply on filters though

Reply:
A similar handlelinks option in filters? 

That's a really interesting idea, but every filter would need to be reworked, that's quite an impact. I'd keep that idea in mind with https://github.com/api-platform/core/issues/2400 for a major overhaul of filters, maybe?